pipeline/Serverless/02_pipeline/combine.py

```diff
@@ -0,0 +1,183 @@
+        archive_path=archive_path,
+        python_version=python_version,
+        license_info=license_info,
+        combined_name=combined_package_name
+    )
+
+    return layers
+
+
+def combined_name(packages: list[str]) -> str:
+    """
+    Args:
+        packages: List of strings of packages to combine
+    returns:
+        combined_name: String of combined package name with dash in between
+        Eg: custom-pandas-tabulate-...
+
+    Example:
+        packages = ["pandas", "tabulate"]
+        combined_name = "custom-pandas-tabulate-..."
+
+    Example:
+        packages = ["pandas", "numpy", "tabulate", "scipy"]
+        combined_name = "custom-pandas-numpy-tabulate-scipy-...
+    """
+    combined_name = "custom"
+    for name in packages:
+        combined_name = combined_name + "-" + name
+
+    logger.info(f"New name: {combined_name}")
+    return combined_name
+
+
+def combine_packages(
+    packages: list[str], python_version: str, combined_name: str
+) -> None:
+    """
+    Args:
+        packages: List of strings of packages to combine
+        python_version: String of python version
+        combined_name: Name of combined package
+    returns:
+        combined_name: combined file name to upload
+    """
+
+    archive_path = f"/tmp/{combined_name}.zip"
+
+    for package in packages:
+        s3_key = f"{python_version}/{package}.zip"
+        download_path = f"/tmp/{package}.zip"
+        s3.download_file(os.environ["BUCKET_NAME"], s3_key, download_path)
+        logger.info(f"Download {s3_key} to {download_path}")
+        # Unzip file
+        with ZipFile(download_path, "r") as zip_object:
+            zip_object.extractall("/tmp")
+
+    result = shutil.make_archive(
+        base_name=archive_path.split(".")[0],  # remove the .zip
+        format="zip",
+        base_dir="python",
+        root_dir="/tmp",
+    )
+
+    return archive_path
+
+
+def upload_to_s3(archive_path: str, s3_location: str):
+    """
+    Args:
+        archive_path: Location of zip file to be uploaded to S3 bucket
+        combined_name: Name of combined package
+    returns:
+        uploaded_file_name: Name of file in S3 bucket
+    """
+    bucket_name = os.environ["BUCKET_NAME"]
+
+    logger.info(
+        f"Uploading {archive_path} with name {s3_location} to S3 bucket {bucket_name}"
+    )
+    s3.upload_file(archive_path, bucket_name, s3_location)
+
+    return None
+
+
+def publish_layer(regions: list[str], archive_path: str, python_version: str, combined_name: str, license_info: str):
+    """
+    Args:
+        regions: List of regions to deploy to
+        archive_path: Location of zip file to be uploaded to S3 bucket
+        combined_name: Name of combined package
+    returns:
+        layer_arns: List of layer ARNs deployed
+    """
+
+
+    layer_name = (
+        f"{os.environ['LAMBDA_LAYER_PREFIX']}{python_version.replace('.','')}-{combined_name}"
+    )
+
+    with open(archive_path, "rb") as zip_file:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using potentially untrusted inputs to access a file path. To protect your code from a path traversal attack, verify that your inputs are sanitized. Learn more about path traversal vulnerabilities on the [Common Weakness Enumeration](https://cwe.mitre.org/data/definitions/22.html) website and the [Open Web Application Security Project](https://owasp.org/www-community/attacks/Path_Traversal) website.